front/src/modules/pathfinding/components/Itinerary/DisplayItinerary/Origin.tsx

```diff
@@ -12,9 +13,10 @@ import { makeEnumBooleans } from 'utils/constants';
 
 type OriginProps = {
   zoomToFeaturePoint: (lngLat?: Position, id?: string) => void;
+  invalidPathItems?: string[];
```

Comment:
I think it would be more elegant to give isInvalid to each component (Origin, Via and Destination) instead of the whole array invalidPathItems. 
You can create a utils function to check whether a pathItem is invalid or not and then call it in Itinerary.
```suggestion
  isInvalid?: boolean;
```

Reply:
I think we don't have `PathStep`s objects at this point. If one `PathItemLocation` is invalid, we don't have a pathfinding result. We only have `TrainScheduleBase.paths` and nothing else IIRC.

Reply:
The above seems to be false: the frontend strips invalid `PathItemLocation` entries and re-laucnhes pathfinding. So in the end we do have `PathStep`s.

Related: https://github.com/OpenRailAssociation/osrd/pull/9270#discussion_r1806605475

Reply:
It's good for me, @emersion ?

Reply:
Yeah, the new `isInvalid` field sgtm :)